tests/testthat/test-cut.R

```diff
@@ -0,0 +1,177 @@
+    args = list(list(column = I(1:3), type = list(value = list(class = "categorical", 
+        categories = list(list(id = 1L, name = "(-1.5,-0.44]", 
+            numeric_value = NULL, missing = FALSE), list(id = 2L, 
+            name = "(-0.44,0.61]", numeric_value = NULL, missing = FALSE), 
+            list(id = 3L, name = "(0.61,1.7]", numeric_value = NULL, 
+                missing = FALSE))))), list(`function` = "and", 
+        args = list(list(`function` = "<=", args = list(list(
+            value = -1.4998629), list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"))), 
+            list(`function` = "<", args = list(list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"), 
+                list(value = -0.4424))))), list(`function` = "and", 
+        args = list(list(`function` = "<=", args = list(list(
+            value = -0.4424), list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"))), 
+            list(`function` = "<", args = list(list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"), 
+                list(value = 0.6119))))), list(`function` = "and", 
+        args = list(list(`function` = "<=", args = list(list(
+            value = 0.6119), list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"))), 
+            list(`function` = "<", args = list(list(variable = "https://app.crunch.io/api/datasets/1/variables/birthyr/"), 
+                list(value = 1.6693629))))))))
+     
+
+##############################################################################
+   basic_cut <- cut(ds$birthyr, 3, variable.name = "new_var", label = c("one", "two", "three"))
+   expect_is(
+       basic_cut,
+       "VariableDefinition"
+   ) 
+   
+   expect_error( cut(ds$birthyr, 3), "Must provide the name for the new variable")
+   
+   expect_equivalent(
+       basic_output,
+       basic_cut
+   )
+   
+   expect_equivalent(
+       right_false,
+       cut(ds$birthyr,
+           3,
+           variable.name = "new_var",
+           label = c("one", "two", "three"),
+           right = FALSE)
+   )
+   
+   expect_equivalent(
+       set_breaks,
+       cut(ds$birthyr,
+           c(-1.4967, 0, 1.6662),
+           variable.name = "new_var",
+           label = c("one", "two"),
+           right = FALSE)
+   )
+   expect_equivalent(
+       dig_lab,
+       cut(ds$birthyr,
+           3,
+           dig.lab = 2,
+           variable.name = "new_var")
+   )
+   expect_identical(
+       crunch:::generateCutLabels(5, c(2.111111, 3, 4, 5), 4, FALSE, FALSE),
+       c("[2.1111,3)", "[3,4)", "[4,5)")
+   )
+   expect_identical(
+       crunch:::generateCutLabels(2, c(2.111111, 3, 4, 5), 4, FALSE, FALSE),
+       c("[2.1,3)", "[3,4)", "[4,5)")
+   )
+  })
+})
+
+with_test_authentication({
+    ds <- newDataset(df)
+    
+    test_that("cut returns the same thing for crunch variables and identical vectors", {
```

Comment:
Please break this into multiple `test_that`s with informative labels that explain what each test case is asserting differently.